importlib_metadata/compat/stdlib.py

```diff
@@ -0,0 +1,27 @@
+"""
+Compatibility layer with stdlib.
+Only needed when distributing via PyPI/3rd-party package.
+"""
+
+import sys
+from typing import TYPE_CHECKING, List, Union
+
+if TYPE_CHECKING:
+    # Avoid circular imports
+
+    from importlib import metadata as _legacy
+
+    from typing_extensions import TypeAlias
+
+    from .. import Distribution, PackagePath, _meta
+
+    if sys.version_info >= (3, 10):
+        from importlib.metadata import PackageMetadata as _legacy_Metadata
+    else:
+        from email.message import Message as _legacy_Metadata
+
+    _PackageMetadataOrLegacy: TypeAlias = Union[_legacy_Metadata, _meta.PackageMetadata]
```

Comment:
I previously tried a more modern syntax, but apparently `from __future__ import annotations` does not work when defining aliases (i.e. you cannot have `|` or `list[...]` on the right side of an assignment).

https://github.com/python/importlib_metadata/actions/runs/9643043347/job/26592160998?pr=487

Reply:
Sometimes in this case, I would suggest splitting the definition on Python version, but given that anyone could try to implement the more modern syntax and immediately get feedback that it's invalid on Python 3.9, I'm perfectly fine with just leaving it as-is. It might mean that this syntax lingers longer than it has to after dropping support for Python 3.9, but that's an acceptable tradeoff for having a unified definition.